### PR TITLE
Add a couple more simple backup tests

### PR DESCRIPTION
**Pull request by api-Hypernova**
Added a few negative backup tests to the minikube test suite (mainly by messing with the SMs in various ways). The test coverage is by no means complete and needs additional work to design and implement more negative and edge-case tests.
The indentation is a bit of a mess but I can't figure out how to fix it - perhaps the original code was indented with tabs, but my editor is replacing it with whitespace. Will try to fix in an upcoming commit.

### CODE REVIEW

File: test/minikube/minikube_base_database_test.go [hunk 1]
```diff
@@ -16,6 +16,10 @@ import (
 	"github.com/gruntwork-io/terratest/modules/helm"
 	"github.com/gruntwork-io/terratest/modules/k8s"
 	"github.com/gruntwork-io/terratest/modules/random"
+
+        //metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+        //"os/exec"
```
Comment (vegichan) [inline test/minikube/minikube_base_database_test.go]: remove these please

File: test/minikube/minikube_base_database_test.go [hunk 2]
```diff
@@ -345,7 +360,7 @@ func TestKubernetesBackupDatabase(t *testing.T) {
 		populateCreateDBData(t, namespaceName, admin0)
 
 		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
-		backupDatabase(t, namespaceName, admin0, "demo", &databaseOptions)
+                backupDatabase(t, namespaceName, admin0, "demo", false, &databaseOptions)
```
Comment (vegichan) [inline test/minikube/minikube_base_database_test.go]: the space is screwed up here and below

**Comment by vegichan** [comment]
I will grab your branch and fix the large number of spacing changes

File: test/testlib/minikube_utilities.go [hunk 3]
```diff
@@ -513,26 +512,11 @@ func KillAdminPod(t *testing.T, namespace string, podName string) {
 	assert.Assert(t, strings.Contains(output, "deleted"), "`deleted` not found in %s", output)
 }
 
-func KillAdminProcess(t *testing.T, namespace string, podName string) {
+func KillProcess(t *testing.T, namespace string, podName string) {
```
Comment (vegichan) [inline test/testlib/minikube_utilities.go]: this is a really good change! 👍

File: test/minikube/minikube_base_database_test.go [hunk 4]
```diff
@@ -141,11 +141,18 @@ func verifyEngineAltAddress(t *testing.T, namespaceName string, admin0 string, e
...
 	// wait for the backup to both start _and_ complete successfully
+	// if we expect backup to never schedule, don't use await. Instead sleep the backuptimeout and check the state of the pod using the same
+	backupTimeout := 120*time.Second
 	backupJob := fmt.Sprintf("hotcopy-%s-job-initial", databaseName)
-	testlib.AwaitPodPhase(t, namespaceName, backupJob, corev1.PodSucceeded, 120*time.Second)
+	if(!shouldTimeout) {
+		testlib.AwaitPodPhase(t, namespaceName, backupJob, corev1.PodSucceeded, backupTimeout)
+	} else {
+		testlib.AwaitPodPhase(t, namespaceName, backupJob, corev1.PodSucceeded, backupTimeout)
+	}
```
Comment (vegichan) [inline test/minikube/minikube_base_database_test.go]: this is the same code. It does not match what you wrote in the comment. What should it do?

File: test/minikube/minikube_base_database_test.go [hunk 5]
```diff
@@ -369,11 +380,102 @@ func TestKubernetesBackupDatabase(t *testing.T) {
...
+				"database.sm.noHotCopy.replicas":        "1",
+			},
+		}
+
+		databaseChartName := testlib.StartDatabase(t, namespaceName, admin0, &databaseOptions)
+		opts := k8s.NewKubectlOptions("", "")
+		opts.Namespace = namespaceName
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		smPodName := fmt.Sprintf("sm-%s-nuodb-%s-%s-hotcopy-0", databaseChartName, opt.ClusterName, opt.DbName)
+		fmt.Printf(k8s.RunKubectlAndGetOutputE(t, opts, "label", "--overwrite", "pods", "-n", namespaceName, smPodName, "role=nohotcopy"))
```
Comment (vegichan) [inline test/minikube/minikube_base_database_test.go]: dynamic labeling is not picked up. This code does nothing. Please remove it.

File: test/minikube/minikube_base_database_test.go [hunk 6]
```diff
@@ -369,11 +380,102 @@ func TestKubernetesBackupDatabase(t *testing.T) {
...
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		ss := fmt.Sprintf("sm-%s-nuodb-%s-%s-hotcopy", databaseChartName, opt.ClusterName, opt.DbName)
+
+		fmt.Printf(k8s.RunKubectlAndGetOutputE(t, opts, "scale", "-n", namespaceName, "statefulset", ss, "--replicas=0"))
+		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
+		backupDatabase(t, namespaceName, admin0, "demo", true, &databaseOptions)
+	})
+
+
+	t.Run("killSMProcessTest", func(t *testing.T) {
```
Comment (vegichan) [inline test/minikube/minikube_base_database_test.go]: what does this test do?

File: test/minikube/minikube_base_database_test.go [hunk 7]
```diff
@@ -369,11 +380,102 @@ func TestKubernetesBackupDatabase(t *testing.T) {
...
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		smPodName := fmt.Sprintf("sm-%s-nuodb-%s-%s-hotcopy-0", databaseChartName, opt.ClusterName, opt.DbName)
+		fmt.Printf(k8s.RunKubectlAndGetOutputE(t, opts, "label", "--overwrite", "pods", "-n", namespaceName, smPodName, "role=nohotcopy"))
+		populateCreateDBData(t, namespaceName, admin0)
+		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
+		backupDatabase(t, namespaceName, admin0, "demo", false, &databaseOptions)
 	})
+
+
+	t.Run("backupNoSMs", func(t *testing.T) {
```
Comment (vegichan) [inline test/minikube/minikube_base_database_test.go]: this test will be flaky, since the backup could be taken before you scale down the SM (which in turn will kill the TE). The better option is to start the database with 1 non-hotcopy SM and 0 hotcopy SMs.

File: test/minikube/minikube_base_database_test.go (hunk 6 above)
Comment (api-Hypernova) [reply test/minikube/minikube_base_database_test.go]: It kills the only backup SM while the initial backup job is in some indeterminate state (probably not started), pauses to let the SM restart and backup to complete. The backup should still verify correctly. Might be able to remove the sleep (the await timeout in backupdatabase might be enough.. will experiment with that).
It is not a super useful test in this current state. It would be better to wait for the initial backup job to enter running state, kill it then, and examine the results. But I have no idea how to validate this kind of situation from an automation point of view, it would need to be manually tested first.

File: test/minikube/minikube_base_database_test.go (hunk 7 above)
Comment (api-Hypernova) [reply test/minikube/minikube_base_database_test.go]: Good idea, will do.

Comment (api-Hypernova) [reply test/minikube/minikube_base_database_test.go]: In this case it becomes exactly the same as the above test where I removed the labeling and is skipped (https://github.com/nuodb/nuodb-helm-charts/issues/64). I will remove this test.

File: test/minikube/minikube_base_database_test.go (hunk 6 above)
Comment (vegichan) [reply test/minikube/minikube_base_database_test.go]: Looking at the logs for this test, it runs for 2 min and does not verify anything. Maybe we can improve on it in the future

File: test/minikube/minikube_base_database_test.go [hunk 8]
```diff
@@ -369,11 +374,68 @@ func TestKubernetesBackupDatabase(t *testing.T) {
...
+
+		defer testlib.Teardown(testlib.TEARDOWN_BACKUP)
+		opt := testlib.GetExtractedOptions(&databaseOptions)
+		opts := k8s.NewKubectlOptions("", "")
+		opts.Namespace = namespaceName
+
+		smPodName := fmt.Sprintf("sm-%s-nuodb-%s-%s-hotcopy-0", databaseChartName, opt.ClusterName, opt.DbName)
+
+		testlib.KillProcess(t, namespaceName, smPodName)
+                backupDatabase(t, namespaceName, admin0, "demo", true, &databaseOptions)
```
Comment (vegichan) [inline test/minikube/minikube_base_database_test.go]: spacing is screwed up again!

File: test/minikube/minikube_base_database_test.go (hunk 4 above)
Comment (api-Hypernova) [reply test/minikube/minikube_base_database_test.go]: Removed, thx.

File: test/minikube/minikube_base_database_test.go (hunk 6 above)
Comment (api-Hypernova) [reply test/minikube/minikube_base_database_test.go]: Agreed. At least it runs for 1 min now.

**Comment by vegichan** [comment]
This issue is stale because it has been open 3 days with no activity
